hbase-mapreduce/src/main/java/org/apache/hadoop/hbase/mapreduce/WALPlayer.java

```diff
@@ -344,6 +345,9 @@ public Job createSubmittableJob(String[] args) throws IOException {
     try {
       TableMapReduceUtil.addDependencyJarsForClasses(job.getConfiguration(),
         Class.forName(codecCls));
+      // get delegation token for the filesystem
+      TokenCache.obtainTokensForNamenodes(job.getCredentials(), new Path[] { new Path(inputDirs) },
```

Comment:
@mymeiyi 
We do get credentials from namenode [here](https://github.com/apache/hadoop/blob/trunk/hadoop-mapreduce-project/hadoop-mapreduce-client/hadoop-mapreduce-client-core/src/main/java/org/apache/hadoop/mapreduce/JobSubmitter.java#L170) 
I don't know much about WALPlayer internals. Does this MR job access 2 different namenodes ?

Reply:
I think JobSubmitter get a token from hdfs for yarn cluster, here we also need a token from hdfs for hbase cluster